contracts/nft-escrow/src/lib.rs

```diff
@@ -0,0 +1,180 @@
+            nonce: payment.token_nonce,
+            wanted_nft,
+            wanted_nonce,
+            wanted_address,
+        };
+
+        self.offers(offer_id).set(offer);
+
+        offer_id
+    }
+
+    #[endpoint]
+    fn cancel(&self, offer_id: u32) {
+        let offers_mapper = self.offers(offer_id);
+
+        require!(!offers_mapper.is_empty(), "Offer does not exist");
+
+        let caller = self.blockchain().get_caller();
+
+        let offer = offers_mapper.get();
+
+        require!(
+            offer.creator == caller,
+            "Only the offer creator can cancel it"
+        );
+
+        self.created_offers(&caller).swap_remove(&offer_id);
+        self.wanted_offers(&offer.wanted_address)
+            .swap_remove(&offer_id);
+
+        self.offers(offer_id).clear();
+
+        self.send().direct_esdt(
+            &offer.creator,
+            &offer.nft,
+            offer.nonce,
+            &BigUint::from(1u64),
+        );
+    }
+
+    #[payable("*")]
+    #[endpoint]
+    fn accept(&self, offer_id: u32) {
+        let offers_mapper = self.offers(offer_id);
+
+        require!(!offers_mapper.is_empty(), "Offer does not exist");
+
+        let offer = offers_mapper.get();
+
+        let caller = self.blockchain().get_caller();
+
+        require!(offer.wanted_address == caller, "Can not accept this offer");
+
+        let payment = self.call_value().single_esdt();
+
+        require!(
+            payment.token_identifier == offer.wanted_nft
+                && payment.token_nonce == offer.wanted_nonce
+                && payment.amount == 1,
+            "NFT does not match"
+        );
+
+        self.created_offers(&offer.creator).swap_remove(&offer_id);
+        self.wanted_offers(&offer.wanted_address)
+            .swap_remove(&offer_id);
+
+        self.offers(offer_id).clear();
+
+        self.send().direct_esdt(
+            &offer.creator,
+            &payment.token_identifier,
+            payment.token_nonce,
+            &payment.amount,
+        );
+        self.send().direct_esdt(
+            &offer.wanted_address,
+            &offer.nft,
+            offer.nonce,
+            &BigUint::from(1u64),
+        );
+    }
+
+    #[view(getCreatedOffers)]
+    fn get_created_offers(
+        &self,
+        address: ManagedAddress,
+    ) -> MultiValueEncoded<MultiValue2<u32, Offer<Self::Api>>> {
+        let mut result = MultiValueEncoded::new();
+
+        for offer_id in self.created_offers(&address).iter() {
+            result.push(MultiValue2::from((offer_id, self.offers(offer_id).get())));
+        }
+
+        result
+    }
+
+    #[view(getWantedOffers)]
+    fn get_wanted_offers(
+        &self,
+        address: ManagedAddress,
+    ) -> MultiValueEncoded<MultiValue2<u32, Offer<Self::Api>>> {
+        let mut result = MultiValueEncoded::new();
+
+        for offer_id in self.wanted_offers(&address).iter() {
+            result.push(MultiValue2::from((offer_id, self.offers(offer_id).get())));
```

Comment:
```
let offer = self.offers(offer_id).get()
let offer_result = MultiValue2::from((offer_id, offer));
result.push(offer_result);
```
more verbose, but easier to read and review

Reply:
if I think better of it extract it into a function and deduplicate the code from `get_created_offers` as well